src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTagExpressionError.java

```diff
@@ -12,7 +12,7 @@
 
 /**
  * Testing tag writing
- * @author Daniel Grenner  // violation
+ * @author Daniel Grenner  // violation 'The tagFormat requires ABC, not Daniel Grenner.'
```

Comment:
move all trailing comments out of javadoc comments. use `// violation 7 lines above|below `

Reply:
It would seem that whenever a comment contains the word "// violation", the line is considered to be an expected value, thus causing an error. But if I just wrote "//" plus any character other than "violation", there would be no error.
In this case, if I had used // violation 7 lines above, the expected value would have been reduced by 7 lines accordingly, thus causing an error.

Reply:
this Check validate comment content, do not change comment content by such technical markers.

Reply:
/**
 * Testing tag writing
 * @author Daniel Grenner  // violation
 
If I change the above code to the following code:
 
// violation 3 lines below 'The tagFormat requires ABC, not Daniel Grenner.'
 /**
 * Testing tag writing
 * @author Daniel Grenner 
 
The test will say:
 Actual and expected violations differ.
 expected to match: 15:(?:\d+:)?\s.*The tagFormat requires ABC, not Daniel Grenner..* 
 but was: 15: The tag @author must match: 'ABC'.
 
 Do you have any idea how to solve this problem?

Reply:
Keep debugging test execution

Reply:
final String[] expected = {
            "15: " + getCheckMessage(MSG_TAG_FORMAT, "@author", "ABC")
        };
        
According to the expected array in the test method, the violation comment can only match an actual error in Chinese, but not in English. What should I do?

// violation 3 lines below '标签 @author 必须匹配： 'ABC' 。'
/**
 * Testing tag writing
 * @author Daniel Grenner
 (success!)
--------------------------------------------------------------------------------------------------------------
 // violation 3 lines below '@author must match 'ABC'.'
/**
 * Testing tag writing
 * @author Daniel Grenner
(failure)
 
 Actual and expected violations differ.
expected to match: 15:(?:\d+:)?\s.*@author must match 'ABC'..*
but was          : 15: 标签 @author 必须匹配： 'ABC' 。

Reply:
Ok, we run into locale problem https://github.com/checkstyle/checkstyle/issues/12104

Please enforce English locale for test execution.
Sorry for inconvenience

Reply:
I see. Thank you